pkg/daemon/daemon.go

```diff
@@ -322,7 +327,16 @@ func (dn *Daemon) checkFiles(files []ignv2_2types.File) bool {
 // checkFileContents reads the file from the filepath and compares its contents
 // with the  expectedContent. It logs an error in case of an error or mismatch
 // and returns the status of the evaluation.
-func checkFileContents(filePath, expectedContent string) bool {
+func checkFileContentsAndMode(filePath, expectedContent string, mode os.FileMode) bool {
+	fi, err := os.Lstat(filePath)
+	if err != nil {
+		glog.Errorf("could not read file: %q, error: %v", filePath, err)
```

Comment:
might be better if it's `could not stat file` or `could not get file info` or something like that.